content/en/blog/_posts/2020-03-30-topology-manager-beta.md

```diff
@@ -0,0 +1,513 @@
+
+
+Notice that the call to **<code>GetTopologyHints()</code> **returns a <strong><code>map[string][]TopologyHint</code></strong>. This allows a single <strong><code>HintProvider</code></strong> to provide hints for multiple resource types instead of just one. For example, the <strong><code>DeviceManager</code></strong> requires this in order to pass hints back for every resource type registered by its plugins.
+
+As **<code>HintProviders</code>** generate their hints, they only consider how alignment could be satisfied for <em>currently</em> available resources on the system. Any resources already allocated to other containers are not considered.
+
+For example, consider the system in Figure 1, with the following two containers requesting resources from it:
+
+
+<table>
+  <tr>
+   <td><strong><code>Container0</code></strong>
+   </td>
+   <td><strong><code>Container1</code></strong>
+   </td>
+  </tr>
+  <tr>
+   <td>
+     
+  ```
+  spec:
+    containers:
+     - name: numa-aligned-container0
+       image: alpine
+       resources:
+       limits:
+          cpu: 2
+          memory: 200Mi
+          gpu-vendor.com/gpu: 1
+          nic-vendor.com/nic: 1
+  ```
+   </td>
+   <td>
+  
+  ```
+  spec:
+    containers:
+     - name: numa-aligned-container1
+       image: alpine
+       resources:
+       limits:
+          cpu: 2
+          memory: 200Mi
+          gpu-vendor.com/gpu: 1
+          nic-vendor.com/nic: 1
+  ```
+   </td>
+  </tr>
+</table>
+
+
+If **<code>Container0</code>** is the first container considered for allocation on the system, the following set of hints will be generated for the three topology-aware resource types in the spec.
+
+	               
+	cpu: {{01: True}, {10: True}, {11: False}}
+	gpu-vendor.com/gpu: {{01: True}, {10: True}}
+	nic-vendor.com/nic: {{01: True}, {10: True}}
+
+With a resulting aligned allocation of:
+
+
+```
+{cpu: {0, 1}, gpu: 0, nic: 0}
+```
+
+![drawing](../../../../static/images/blog/2020-03-25-kubernetes-1.18-release-announcement/numa-hint-provider1.png)
+
+When considering **<code>Container1</code>** these resources are then presumed to be unavailable, and thus only the following set of hints will be generated:
+
+	
+	cpu: {{01: True}, {10: True}, {11: False}}
+	gpu-vendor.com/gpu: {{10: True}}
+	nic-vendor.com/nic: {{10: True}}
+
+With a resulting aligned allocation of:
+
+
+```
+{cpu: {4, 5}, gpu: 1, nic: 1}
+```
+
+![drawing](../../../../static/images/blog/2020-03-25-kubernetes-1.18-release-announcement/numa-hint-provider2.png)
+
+**NOTE: **Unlike the pseudocode provided at the beginning of this section, the call to **<code>Allocate()</code>** does not actually take a parameter for the merged “best” hint directly. Instead, the <strong><code>TopologyManager</code></strong> implements the following <strong><code>Store</code></strong> interface that <strong><code>HintProviders</code></strong> can query to retrieve the hint generated for a particular container once it has been generated:
```

Comment:
```suggestion
**NOTE:**Unlike the pseudocode provided at the beginning of this section, the call to **<code>Allocate()</code>** does not actually take a parameter for the merged “best” hint directly. Instead, the <strong><code>TopologyManager</code></strong> implements the following <strong><code>Store</code></strong> interface that <strong><code>HintProviders</code></strong> can query to retrieve the hint generated for a particular container once it has been generated:
```

I believe the space after NOTE: is breaking formatting, the text is coming across as
```
** NOTE: **
```

Reply:
Thank you. Fixed in the latest version